cpp/ql/lib/semmle/code/cpp/controlflow/IRGuards.qll

```diff
@@ -125,6 +125,20 @@ class GuardCondition extends Expr {
   cached
   predicate ensuresLt(Expr left, Expr right, int k, BasicBlock block, boolean isLessThan) { none() }
 
+  /**
+   * Holds if (determined by this guard) `e < k` evaluates to `isLessThan` if
+   * this expression evaluates to `value`.
+   */
+  cached
+  predicate comparesLt(Expr e, int k, boolean isLessThan, AbstractValue value) { none() }
+
+  /**
+   * Holds if (determined by this guard) `e < k` must be `isLessThan` in `block`.
+   * If `isLessThan = false` then this implies `left >= k`.
```

Comment:
`left` seems out of place here.

Reply:
Good point. I think it's a copy/paste error. Will fix!

Reply:
Fixed in 3a7b80da47417183f73fe68b67ba0c647ab65c8f